### PR TITLE
tournament different publishing option

### PR DESCRIPTION
**Pull request by smallst**
Now you can either select `review results`, when tournament finish `ranking`, it goes into `waiting` state, and email owner he can review, then owner publish the tournament rankings.
image
or `publish immediately`, when tournament finish `ranking`, it goes into `ended` (published) directly,
image
or select a determine date to publish, when tournament finish `ranking`, it goes into `waiting` state, and when resultsDate coming, it published. (Notice: the tournament.state doesn't update in time, while when a player visit the tournament, it can get the correct state, so it works well)
image

when `review` tournament is visited by owner: 
image
when `review` tournament is visited by others:
image

### CODE REVIEW

File: app/views/ladder/MainLadderViewV2.vue [hunk 1]
```diff
@@ -222,14 +222,16 @@ export default {
...
           resultsDate: moment().add(3, 'day').toISOString(),
+          waiting: false,
           editing: 'new'
         }
         this.showModal = true
       }
     },
     handleEditTournament (tournament) {
-      /* console.log('handle edit', tournament) */
+      console.log('handle edit', tournament)
```
Comment (mrfinch) [inline app/views/ladder/MainLadderViewV2.vue:232]: remove

File: app/views/ladder/LadderView.coco.coffee [hunk 2]
```diff
@@ -282,6 +283,17 @@ module.exports = class LadderView extends RootView
   onClickPlayButton: (e) ->
     @showPlayModal($(e.target).closest('.play-button').data('team'))
 
+  onClickPublishButton: (e) ->
+    return unless (@tournamentId and @tournamentState == 'waiting' and me.get('_id') == @league.ownerID)
+    $.ajax
```
Comment (mrfinch) [inline app/views/ladder/LadderView.coco.coffee:288]: how about adding it in app/core/api/tournaments.js like we do for others?

**Review by mrfinch — COMMENTED** [review]
Can you add screenshots of the change? It always helps to have gif/screenshot for visual changes in a PR.

File: app/locale/en.coffee [hunk 3]
```diff
@@ -4200,7 +4200,10 @@ module.exports = nativeDescription: "English", englishDescription: "English", tr
     my_teams: 'My Teams:'
     team: 'Team'
     team_description: '(Required) Only the players in the team can join this tournament'
+    waiting_publish: 'Review Results'
+    waiting_description: 'We will email you the results of tournament, and wait you to publish it to participants.'
```
Comment (nwinter) [inline app/locale/en.coffee:4204]:
```suggestion
    waiting_description: 'We will email you the results of tournament and wait for you to publish for participants.'
```

**Comment by smallst** [comment]
when test locally. since there is not bgp, so you need to manually update database to change state to `waiting` to check `publish` button.

**Pull request merged**